Add: implement custom https header based whitelisting (#242)
10up/restricted-site-access: README.md

  * [Is there a way to configure this with WP-CLI?](#is-there-a-way-to-configure-this-with-wp-cli)
  * [How can I programmatically define whitelisted IPs?](#how-can-i-programmatically-define-whitelisted-ips)
  * [Is there a constant to control my site restriction?](#is-there-a-constant-i-can-set-to-ensure-my-site-is-or-is-not-restricted)

>   * [Can I access website based on whitelist custom https headers?](#can-i-access-website-based-on-whitelist-custom-https-headers)

Reviewer: ```suggestion
  * [Can I provide access to my site based on custom HTTP headers?](#can-i-provide-access-to-my-site-based-on-custom-http-headers)
```

If your proxy does not use static IP addresses, you can still utilize the `rsa_trusted_headers` filter to change which HTTP headers you want to trust.

> You can also use `rsa_custom_trusted_headers` filter hooks to allow request with custom headers.

Reviewer: ```suggestion
You can also use the `rsa_custom_trusted_headers` filter to allow a request with custom headers.
```

> All custom headers should be present in request with unique value. You can change this value if compromised to protect your site.

Reviewer: ```suggestion
All custom headers should be present in the request and with unique values. You can change these values if compromised to protect your site.
```

Please note that setting `RSA_FORCE_RESTRICTION` will override `RSA_FORBID_RESTRICTION` if both are set.

> ### Can I access website based on whitelist custom https headers?

Reviewer: ```suggestion
### Can I provide access to my site based on custom HTTP headers?
```

> You can use the `restricted_site_access_is_restricted` filter hook to allow site access to whitelist custom headers.

Reviewer: ```suggestion
You can use the `restricted_site_access_is_restricted` filter hook to allow access based on custom headers.
```

> The custom header you want to allow should be present in the request with a unique value. You can allow more than one custom header.

Reviewer: ```suggestion
The custom header you want to allow should be present in the request and should contain a unique value. If needed, you can allow more than one header.
```

> Whitelist custom header should be present in the request with unique value. You can whitelist more than one custom trusted header.

Reviewer: ```suggestion
The custom header you want to allow should be present in the request with a unique value. You can allow more than one custom header.
```

> You can change these values if compromised to protect your site.

Reviewer: ```suggestion
If these header/value pairs are ever compromised, you should change the accepted values in order to protect your site.
```

> Add custom trusted header in `$allowed_custom_trusted_headers` in code snippet and Place the following PHP code in the theme's functions.php file or in a simple plug-in:

Reviewer: ```suggestion
See below for an example code snippet you can utilize:
```

```php
<?php
/**
 * Add custom trusted header validation.
 *
 * IP restriction will be bypassed if the trusted custom header is present and has the correct value.
 */

> add_filter( 'restricted_site_access_is_restricted', function ( $is_restricted ){

Reviewer: ```suggestion
add_filter( 'restricted_site_access_is_restricted', function ( $is_restricted ) {
```

	// Custom trusted headers; array key should be the header name and value should be the header value.
	$allowed_custom_trusted_headers = array(

> 		'HTTP_RSA_CUSTOM_HEADER' => 'value' // Replace value with the header value.

Reviewer: ```suggestion
		'HTTP_RSA_CUSTOM_HEADER' => 'value' // Replace header and value with your custom details.
```
